app/models/host/base.rb

```diff
@@ -234,6 +235,14 @@ def set_interfaces(parser)
       self.interfaces.reload
     end
 
+    def set_reported_data(parser)
+      if parser.boot_time
+        reported_data_facet = self.reported_data || self.build_reported_data
```

Comment:
Maybe extract this to a method so you can just call `host.reported_data_facet` and get either the saved object or a new one?
Then we could just do `reported_data_facet.update(boot_time: parser.boot_time)`